GSA-TTS/FAC
# Audit Info Form Frontend

# Audit Info Form Frontend - Django View/HTML/JS

Issue #1278

## Changes
1. New `AuditInfoFormView` under `audit` that handles the getting/posting of audit info form. 
2. New HTML `audit-info-form.html`, which is a formatted form that should match very closely to the associated [Figma doc](https://www.figma.com/file/1bNncVt1TcFFDhQslqFC2o/FAC-Playground?node-id=4954%3A24748&mode=dev). 
3. New JS file to handle requiring of checkboxes. HTML5 doesn't have a good way to 'require' multiple choice selections (that thing that jumps you to an incomplete question when you don't answer it). So, I use JS to require/unrequire to mimic the correct behavior. 

Form validation is handled in Django. Unanswered questions will be jumped to if you try to submit without answering them. All questions available in the form are required, so you have to answer everything. 

## How to test
Switch to this branch and run normally. These changes should not require a rebuild, but may require you to restart your container to recognize the JS. Navigate to your [audit home page](http://localhost:8000/audit/submission-progress/YOURIDHERE), and click "[Upload the audit report PDF](http://localhost:8000/audit/audit-info/YOURIDHERE)". Fill out the form. Successful entries will redirect you to your audit home page, and will not currently save anything to the DB.

## Screenshots:
![image](https://github.com/GSA-TTS/FAC/assets/91098850/b1687048-71ca-45f4-8eff-4ce8138123f5)
![image](https://github.com/GSA-TTS/FAC/assets/91098850/d95cc40a-9275-4fbe-8d9f-99894af917c1)
